include/nil/crypto3/zk/snark/arithmetization/plonk/copy_constraint.hpp

```diff
@@ -35,8 +37,74 @@ namespace nil {
             namespace snark {
 
                 template<typename FieldType>
-                using plonk_copy_constraint = std::pair<plonk_variable<typename FieldType::value_type>, plonk_variable<typename FieldType::value_type>>;
+                struct plonk_copy_constraint {
+                    plonk_copy_constraint() = default;
+                    plonk_copy_constraint(const plonk_copy_constraint<FieldType> &other){
+                        initialize(other.first, other.second);
+                    }
+                    plonk_copy_constraint(
+                        const plonk_variable<typename FieldType::value_type> &_first,
+                        const plonk_variable<typename FieldType::value_type> &_second
+                    ){
+                        initialize(_first, _second);
+                    }
+                    plonk_variable<typename FieldType::value_type> first;
+                    plonk_variable<typename FieldType::value_type> second;
+                    bool operator==(const plonk_copy_constraint<FieldType> &other){
+                        return ((first == other.first ) && (second == other.second));
+                    }
+                protected:
+                    void initialize(
+                        const plonk_variable<typename FieldType::value_type> &_first,
+                        const plonk_variable<typename FieldType::value_type> &_second
+                    ){
+                        BOOST_ASSERT(_first.relative == false);
+                        BOOST_ASSERT(_second.relative == false);
+                        if(_first.type == _second.type){
```

Comment:
This is not related to this PR, I needed to notice this in Elena's change but did not. Probably she wanted to create an operator< for class Plonk_variable, and wanted to say 

first = plonk_variable<typename FieldType::value_type>(_first);
second = plonk_variable<typename FieldType::value_type>(_second);

if (_first > _second) 
swap(first, second)

Please fix this in sigle-threaded version as well, send to Elena for review.

Reply:
https://github.com/NilFoundation/crypto3-zk/pull/322